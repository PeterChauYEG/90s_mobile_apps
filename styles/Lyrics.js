// react native
import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  body: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  }
})

export default styles
