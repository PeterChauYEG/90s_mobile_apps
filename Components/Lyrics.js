/**
 * @format
 * @flow
 */

// react
import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

// react native
import { Button, Text } from 'react-native'

// styles
import styles from './styles/Loading'

type Props = {
  clearLyrics: () => void,
  lyrics: string,
  nChars: number,
  sample: string
}

const Lyrics = (props: Props) => {
  const { clearLyrics, lyrics, nChars, sample } = props

  return (
    <Fragment>
      <Text style={styles.body}>Initial Lyrics</Text>
      <Text style={styles.body}>{sample}</Text>

      <Text style={styles.body}>Number of characters generated</Text>
      <Text style={styles.body}>{nChars}</Text>

      <Text style={styles.body}>Generated Lyrics</Text>
      <Text style={styles.body}>{lyrics}</Text>

      <Button
        accessibilityLabel='Start this process over and enter new parameters'
        color='#841584'
        onPress={() => clearLyrics()}
        title='Start over'
      />
    </Fragment>
  )
}

Lyrics.propTypes = {
  clearLyrics: PropTypes.func.isRequired,
  lyrics: PropTypes.string.isRequired,
  nChars: PropTypes.number.isRequired,
  sample: PropTypes.string.isRequired
}
Lyrics.defaultProps = {}

export default Lyrics
