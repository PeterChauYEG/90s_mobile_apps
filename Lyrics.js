/**
 * @format
 * @flow
 */

// react
import React, { Fragment } from 'react'

// react native
import { Text } from 'react-native'

// styles 
import styles from './styles/Loading'

const Lyrics = props => {
  const { lyrics, nChars, sample } = props

  return (
    <Fragment>
      <Text style={styles.body}>Initial Lyrics</Text>
      <Text style={styles.body}>{sample}</Text>
      
      <Text style={styles.body}>Number of characters generated</Text>
      <Text style={styles.body}>{nChars}</Text>
      
      <Text style={styles.body}>Generated Lyrics</Text>
      <Text style={styles.body}>{lyrics}</Text>
    </Fragment>
  )
}

export default Lyrics