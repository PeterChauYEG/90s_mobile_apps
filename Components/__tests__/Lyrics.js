import React from 'react'

// Testing libraries
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import renderer from 'react-test-renderer'

// Component
import Lyrics from '../Lyrics'

// Configure enzyme to work with react
Enzyme.configure({ adapter: new Adapter() })

describe('ErrorOccurred', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Lyrics clearLyrics={() => {}} 
          lyrics='' 
          nChars={0} 
          sample='' 
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('simulates "Start over" button click', () => {
    const clearLyrics = jest.fn()

    const wrapper = shallow(
      <Lyrics clearLyrics={clearLyrics} 
        lyrics='' 
        nChars={0} 
        sample=''
      />
    )
    wrapper.find('Button').simulate('press') 
    expect(clearLyrics).toHaveBeenCalled()
  })
})
