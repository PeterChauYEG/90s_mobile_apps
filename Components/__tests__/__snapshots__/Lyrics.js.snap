// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics renders correctly 1`] = `
Array [
  <Text>
    Initial Lyrics
  </Text>,
  <Text>
    
  </Text>,
  <Text>
    Number of characters generated
  </Text>,
  <Text>
    0
  </Text>,
  <Text>
    Generated Lyrics
  </Text>,
  <Text>
    
  </Text>,
  <View
    accessibilityLabel="Start this process over and enter new parameters"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Start over
      </Text>
    </View>
  </View>,
]
`;
